CMakeLists.txt

```diff
@@ -95,10 +98,30 @@ zephyr_compile_options(
   -Wformat
   -Wformat-security
   -Wno-format-zero-length
+  -imacros ${AUTOCONF_H}
+)
+
+
+# FIXME
+if(CONFIG_ARCH_POSIX)
+zephyr_compile_options(
+  -Wno-undef
+  -Wno-implicit-function-declaration
+  )
+else()
```

Comment:
We need to try to get rid of this else.

-Wno-main can be applied to POSIX and non-POSIX alike I figure, it's just a warning.

I have verified that keeping -ffreestanding here and adding -fno-freestanding to posix/CMakeLists.txt gives us the desired behaviour.